languages/rust/reference/exercise-concepts/paasio.md

```diff
@@ -0,0 +1,187 @@
+# Concepts of Paasio
+
+PaaS I/O explores extending standard `Read` and `Write` traits to provide additional statistics via a wrapper type. A good way to learn about traits in Rust and their best features.
+
+This exercise can be implemented using either a single or two (for `Read` and `Write`)
+newtype structs, with appropriate trait bounds.
+
+## Extracted concepts
+
+- **Primitives**
+  - `usize`
+  - `u8`
+- **Immutability/explicit mutability**
+- **Structs**
+- **Functions**
+  - **Methods**
+- **Slices**
+- **Type aliases**
+- **`Result`** type
+- **`Self`** type
+- **References**
+  - Dereferencing
+- **Traits**
+  - Trait methods
+  - `self` and mutability
+  - Trait bounds
+- **`impl` blocks**
+- `new()` - common practice
+- **Iterators**
```

Comment:
None of the solutions here actually use iterators; they use `map` on results, instead. Suggest removing this concept.